### PR TITLE
Disabling OldBodies/Receipts for non validator nodes

### PR DESCRIPTION
**Pull request by deffrian**
Fixes | Closes | Resolves #
#4818 

## Changes:
- New parameter Sync.NoValidatorLightClient. If set to true user may disable old bodies and receipts

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [] No

**Comments about testing , should you have some** (optional)

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by LukaszRozmej** [comment]
Why is this a PR?

**Comment by deffrian** [comment]
> Why is this a PR?

It's draft

**Comment by LukaszRozmej** [comment]
> > Why is this a PR?
> 
> It's draft, just for testing

No need to be a PR and cause confusion then

File: src/Nethermind/Nethermind.Blockchain/Synchronization/ISyncConfig.cs [hunk 1]
```diff
@@ -97,5 +97,8 @@ public interface ISyncConfig : IConfig
 
         [ConfigItem(Description = "Disable some optimization and run a more extensive sync. Useful for broken sync state but normally not needed", DefaultValue = "false")]
         public bool StrictMode { get; set; }
+
+        [ConfigItem(Description = "[EXPERIMENTAL] Only for non validator nodes! If set to true, DownloadReceiptsInFastSync and/or DownloadBodiesInFastSync can be set to false.", DefaultValue = "false")]
+        public bool NoValidatorLightClient { get; set; }
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Blockchain/Synchronization/ISyncConfig.cs:102]: don't use LightClient for this

**Review by MarekM25 — CHANGES_REQUESTED** [review]
No tests

**Review by MarekM25 — APPROVED** [review]

**Comment by benaadams** [comment]
@deffrian is this working correctly or am I using wrong setting? 🤔

image


image

**Comment by deffrian** [comment]
@benaadams, you should also disable old bodies/receipts `Sync.Download[Bodies/Receipts]InFastSync=false`

**Comment by benaadams** [comment]
> @benaadams, you should also disable old bodies/receipts `Sync.Download[Bodies/Receipts]InFastSync=false`

Perfect, works like a dream 👍

**Pull request merged**